server/src/omtLinkProvider.ts

```diff
@@ -0,0 +1,173 @@
+        return new Map<string, string>(mapSource);
+    }
+}
+
+/**
+ * returns all the text of a line in the provided document
+ * @param document A text document with at least [line] amount of lines
+ * @param line the line to be returned
+ */
+function getLine(document: TextDocument, line: number): string {
+    return document.getText(Range.create(Position.create(line, 0), Position.create(line, Number.MAX_VALUE)));
+}
+
+/**
+ * Makes an absolute path based on a link relative to a document
+ * @param document The document the link is relative to
+ * @param link a relative path from the document
+ */
+function toAbsolutePath(document: TextDocument, link: string): string {
+    return resolve(dirpath(document.uri), link)
+}
+
+/**
+ * Convert the uri to a path to a directory and make sure it does not start with file:
+ * @param uri path to be made into a dirpath
+ */
+function dirpath(uri: string): string {
+    uri = dirname(uri);
+    return uri.startsWith('file:') ? uri.substr(7) : uri;
+}
+
+/**
+ * Replace the start of the uri with the mapped value if it starts with a shorthand, otherwise returns the same path.
+ * @param uri the path that may start with a shorthand
+ * @param shorthands a mapping of shorthands to the paths they represent
+ */
+function replaceStart(uri: string, shorthands: Map<string, string>): string {
+    shorthands.forEach((value, key) => {
+        if (uri.startsWith(key)) {
+            uri = uri.substr(key.length);
+            uri = resolve(value.substr(0, value.lastIndexOf('/*')), '.' + uri);
+        }
+    });
+    return uri;
+}
+
+/**
+ * Find links to other imported OMT files
+ * @param document the OMT file we are scanning
+ * @param resolveShorthand function to resolve shorthand annotations at the start of import paths
+ */
+function findOMTUrl(document: TextDocument, resolveShorthand: (uri: string) => string, workspaceLookup: WorkspaceLookup): DocumentLink[] {
+    // console.log('server omtLinkProvider.findOMTUrl');
+    let documentLinks: DocumentLink[] = [];
+    // so match after import: until we need any other (\w+): without any preceding spaces
+    let isScanning = false;
+    for (let l = 0; l <= document.lineCount - 1; l++) {
+        const line = getLine(document, l);
+        if (importMatch.exec(line)) {
+            isScanning = true; // start scannning lines for import paths after we enter an import block
+        } else if (isScanning) {
+            if (otherDeclareMatch.exec(line)) {
+                break; // we can stop matching after the import block
+            } else {
+                const uriMatch = omtUriMatch.exec(line);
+                const diMatch = declaredImportMatch.exec(line);
+                if (uriMatch) {
+                    // match[0] is the full match inluding the whitespace of match[1]
+                    // match[1] is the whitespace and optional quotes. both of which we don't want to include in the linked text
```

Comment:
I think you mean `diMatch[2]` instead of `match[1]` of the uriMatch right?
because I'm using all of the ones in uriMatch to determine the length of the link

Reply:
need to redo this one, since the test did not catch a mistake